forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -234,46 +235,44 @@ impl WalletCommands {
                     cli_error_and_die(format!("{address} is not a valid address"), 1);
                 }
 
-                let address = address_result.unwrap();
+                let address = address_result?;
 
-                let message = hex::decode(message).unwrap();
+                let message = hex::decode(message).context("Message has to be a hex string")?;
                 let message = base64::encode(message);
 
                 let response = wallet_sign(
                     (AddressJson(address), message.into_bytes()),
                     &config.client.rpc_token,
                 )
                 .await
-                .map_err(handle_rpc_err)
-                .unwrap();
+                .map_err(handle_rpc_err)?;
                 println!("{}", hex::encode(response.0.bytes()));
+                Ok(())
             }
             Self::Verify {
                 message,
                 address,
                 signature,
             } => {
```

Comment:
Let's parse the `Address`  and use `address.protocol()` to implement this logic more robustly.

Reply:
First is done. By second do you mean usage of `state_account_key`?

Reply:
No, look at how the signature is defined. It can be done much better if you use [protocol](https://docs.rs/fvm_shared/2.0.0/fvm_shared/address/struct.Address.html#method.protocol).

Reply:
Pushed. Hope that's what expected